command_manager.gd

```diff
@@ -129,23 +128,21 @@ func go_to_previous_command() -> void:
 ## Returns to a previous [code]GoTo[/code] command call.
 ## See [ReturnValue] to know possible values for [code]return_value[/code]
 ## argument.
-func return_command(return_value:ReturnValue):
+func return_command(return_value:ReturnValue, return_timeline:bool = false):
 	assert(!_jump_history.is_empty())
-	if return_value == ReturnValue.NO_RETURN:
-		_notify_timeline_end()
-		_disconnect_command_signals(current_command)
-		return
-	
-	var jump_data:Array = _jump_history.pop_back()
-	var history_from:Array = jump_data[ _JumpHistoryData.FROM ]
-	
-	var next_command_idx:int = history_from\
-	[ _HistoryData.COMMAND_INDEX ] + return_value
-	
-	var next_timeline:Timeline =  history_from[ _HistoryData.TIMELINE ]
-	
+
+	var next_command_idx:int
+	var next_timeline:Timeline = current_timeline
+	while next_timeline == current_timeline:
+		var jump_data:Array = _jump_history.pop_back()
```

Comment:
popping until it finds a different timeline? Maybe this can cause an infinite loop if is called inside the main timeline and a go_to was used?